assignments/array-methods.js

```diff
@@ -55,29 +55,65 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
-let fullName = [];
-console.log(fullName);
+let fullNames = [];
+runners.forEach(function(runner){
+    fullNames.push(`${runner.first_name} ${runner.last_name}`);
+    return fullNames;
+});
+console.log(fullNames);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+allCaps = runners.map(function(runner){
+    return runner.first_name.toUpperCase();
+});
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+largeShirts = runners.filter(function(runner) {
+    return runner.shirt_size === "L";
+});
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+ticketPriceTotal = runners.reduce(function(total, runner) {
+    return total += runner.donation;
+}, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+//Taran wants to send a thank you email to all of the runners that donated to the fund raiser.
+//Use an array method to find all of the people who donated, and return an array of those people.
+let donatorEmails = [];
+donatorEmails = runners.filter(function(runner) {
+    return runner.donation > 0;
+});
+console.log(donatorEmails);
 
 // Problem 2
+//Find all of the different companies participating in the 5k. Remove all of the duplicates and return them in an array.
+let companies = [];
+runners.forEach(function(runner) {
+    company = runner.company_name;
+    if (!companies.includes(company)) {
+        companies.push(company);
+    }
+});
+console.log(companies);
 
-// Problem 3
\ No newline at end of file
+// Problem 3
+//The runners were split up into two teams, odds and evens. The odds had the best overall time, and will be mailed a coupon book.
+//Use the runner ids to find out who was on the odd team, and return them in an array.
+let winningTeam = [];
+winningTeam = runners.filter(function(runner) {
+    return runner.id % 2 === 1;
+});
+console.log(winningTeam);
```

Comment:
impressive logic here! honestly didn't know how to find even and odds in an array so thanks for that! :)